File: java/java/src/main/java/org/signal/zkgroup/groups/GroupIdentifier.java
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.zkgroup.groups;

import org.signal.zkgroup.InvalidInputException;
import org.signal.zkgroup.internal.ByteArray;

public final class GroupIdentifier extends ByteArray {

  public static final int SIZE = 32;

  public GroupIdentifier(byte[] contents) throws InvalidInputException {
    super(contents, SIZE);
  }

}

File: node/ts/zkgroup/groups/GroupIdentifier.ts
//
// Copyright 2020-2021 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import ByteArray from '../internal/ByteArray.js';
import { Buffer } from 'node:buffer';

export default class GroupIdentifier extends ByteArray {
  private readonly __type?: never;
  static SIZE = 32;

  constructor(contents: Uint8Array) {
    super(contents, GroupIdentifier.checkLength(GroupIdentifier.SIZE));
  }

  /** Returns the group ID as a base64 string (with padding). */
  toString(): string {
    return Buffer.from(this.contents).toString('base64');
  }
}
